### ISSUE TITLE
Question about the speed of parsing

### ISSUE THREAD

**Issue by LongTengDao** [alethiophile/qtoml#3]
> I wrote it because I found PyTOML too slow.

Hi! I want to know, how slow it is, could make user feel slow? Like 2 seconds for parsing a toml doc, and usually the parsing process should take a imperceptible period of time like 2ms?
And what principle could make that big speed difference between these parsers?

I'm a new toml parser coder (not the author of PyTOML), but I'm caring about the speed of parser, which I didn't care about before.

Thank you!

**Comment by alethiophile** [comment]
I had a use case where I needed to encode and decode TOML documents of some megabytes to tens of megabytes. For those large values, PyTOML was taking around a minute to encode or decode, which was unacceptable. When last I checked it, qtoml would do it in two or three seconds.

I haven't profiled PyTOML or looked at its code, so I can't say why it was that slow. I know most of the optimizations I made to qtoml had to do with taking out string copy operations; if you have a single Python string of megabyte size, it takes a long time to copy it, and normal-style Python string manipulation involves doing a lot of copies, including in places where they aren't really necessary (where the copy is just an intermediate value that gets thrown away). If I had to guess, I would say this is probably why PyTOML was slow.
